editor/plugins/texture_region_editor_plugin.cpp

```diff
@@ -1119,88 +1124,40 @@ TextureRegionEditor::TextureRegionEditor() {
+	return Object::cast_to<Sprite2D>(p_object) || Object::cast_to<Sprite3D>(p_object) || Object::cast_to<NinePatchRect>(p_object) || Object::cast_to<StyleBoxTexture>(p_object) || Object::cast_to<AtlasTexture>(p_object);
 }
 
-void TextureRegionEditorPlugin::make_visible(bool p_visible) {
-	if (p_visible) {
-		texture_region_button->show();
-		bool is_node_configured = region_editor->is_stylebox() || region_editor->is_atlas_texture() || region_editor->is_ninepatch();
-		is_node_configured |= region_editor->get_sprite_2d() && region_editor->get_sprite_2d()->is_region_enabled();
-		is_node_configured |= region_editor->get_sprite_3d() && region_editor->get_sprite_3d()->is_region_enabled();
-		if ((is_node_configured && !manually_hidden) || texture_region_button->is_pressed()) {
-			EditorNode::get_singleton()->make_bottom_panel_item_visible(region_editor);
-		}
-	} else {
-		if (region_editor->is_visible_in_tree()) {
-			EditorNode::get_singleton()->hide_bottom_panel();
-			manually_hidden = false;
-		}
-		texture_region_button->hide();
-		region_editor->edit(nullptr);
-	}
-}
-
-Dictionary TextureRegionEditorPlugin::get_state() const {
-	Dictionary state;
-	state["snap_offset"] = region_editor->snap_offset;
-	state["snap_step"] = region_editor->snap_step;
-	state["snap_separation"] = region_editor->snap_separation;
-	state["snap_mode"] = region_editor->snap_mode;
-	return state;
+void EditorInspectorPluginTextureRegion::_region_edit(Object *p_object) {
+	texture_region_editor->edit(p_object);
 }
 
-void TextureRegionEditorPlugin::set_state(const Dictionary &p_state) {
-	Dictionary state = p_state;
-	if (state.has("snap_step")) {
-		Vector2 s = state["snap_step"];
-		region_editor->sb_step_x->set_value(s.x);
-		region_editor->sb_step_y->set_value(s.y);
-		region_editor->snap_step = s;
-	}
-
-	if (state.has("snap_offset")) {
-		Vector2 ofs = state["snap_offset"];
-		region_editor->sb_off_x->set_value(ofs.x);
-		region_editor->sb_off_y->set_value(ofs.y);
-		region_editor->snap_offset = ofs;
-	}
-
-	if (state.has("snap_separation")) {
-		Vector2 sep = state["snap_separation"];
-		region_editor->sb_sep_x->set_value(sep.x);
-		region_editor->sb_sep_y->set_value(sep.y);
-		region_editor->snap_separation = sep;
-	}
-
-	if (state.has("snap_mode")) {
-		region_editor->_set_snap_mode(state["snap_mode"]);
-		region_editor->snap_mode_button->select(state["snap_mode"]);
+bool EditorInspectorPluginTextureRegion::parse_property(Object *p_object, const Variant::Type p_type, const String &p_path, const PropertyHint p_hint, const String &p_hint_text, const uint32_t p_usage, const bool p_wide) {
+	if ((p_type == Variant::RECT2 || p_type == Variant::RECT2I)) {
+		if (((Object::cast_to<Sprite2D>(p_object) || Object::cast_to<Sprite3D>(p_object) || Object::cast_to<NinePatchRect>(p_object) || Object::cast_to<StyleBoxTexture>(p_object)) && p_path == "region_rect") || (Object::cast_to<AtlasTexture>(p_object) && p_path == "region")) {
```

Comment:
Checking property of `StyleBoxTexture` is not necessary, as `region_rect` is the only Rect2 property in there.